crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -151,46 +164,67 @@ impl LogManagerActor {
     /// - sends request successful (mqtt)
     async fn execute_logfile_request_operation(
         &mut self,
+        topic: &Topic,
         request: &LogUploadCmdPayload,
     ) -> Result<(), LogManagementError> {
-        let log_content = log_manager::new_read_logs(
+        let log_path = log_manager::new_read_logs(
             &self.plugin_config.files,
             &request.log_type,
             request.date_from,
             request.lines.to_owned(),
             &request.search_text,
         )?;
 
-        self.send_log_file_http(log_content, request.tedge_url.clone())
+        let upload_request = UploadRequest::new(
+            &request.tedge_url,
+            Utf8Path::from_path(log_path.as_path()).unwrap(),
+        );
+
+        info!(
+            "Awaiting upload of log type: {} to url: {}",
+            request.log_type, request.tedge_url
+        );
+
+        self.upload_sender
+            .send((topic.name.clone(), upload_request))
             .await?;
 
         Ok(())
     }
 
-    async fn send_log_file_http(
+    async fn process_uploaded_log(
         &mut self,
-        log_content: String,
-        url: String,
+        topic: &str,
+        result: UploadResult,
     ) -> Result<(), LogManagementError> {
-        let req_builder = HttpRequestBuilder::put(&url)
-            .header("Content-Type", "text/plain")
-            .body(log_content);
+        if let Some(mut request) = self.pending_operations.remove(topic) {
+            let topic = Topic::new_unchecked(topic);
```

Comment:
issue: How do we know topic passed here is always valid? I'd rather we use `Topic::new()` and return the error, or take `topic: &Topic` as an argument.

Reply:
The HashMap `self.pending_operations` keep topic names (String) that are waiting for uploads by the upload actor. So, the topic from the hashmap should be always valid.